config/initializers/active_job_select_foodcoop.rb

```diff
@@ -10,13 +10,29 @@ def serialize(arguments)
       end
 
       def deserialize(arguments)
-        FoodsoftConfig.select_multifoodcoop arguments.shift
+        FoodsoftConfig.select_multifoodcoop arguments[0]
         orig_deserialize(arguments)
       end
     end
   end
 end
 
+module ActiveJob
+  module Execution
+    # Store the original `_perform_job` method
+    original_perform_job = instance_method(:_perform_job)
+
+    # Override the `_perform_job` method
+    define_method(:_perform_job) do
+      foodsoft_scope = @arguments.shift
+      FoodsoftConfig.select_foodcoop foodsoft_scope
```

Comment:
i think this works only for multi-coop mode and could be the reason why the tests fail with the default `app_config.yml`

Reply:
Ouch, I'm sorry not to have checked that!

Reply:
that's okay! Any idea how we can mitigate this? 
just a wild guess:
```
      if FoodsoftConfig[:multi_coop_install]  do
        foodsoft_scope = @arguments.shift
        FoodsoftConfig.select_foodcoop foodsoft_scope
      end
```

Reply:
But perhaps it's not a good idea to read the config early on (it may involve database access, but didn't check).
The easiest solution may be
```ruby
foodsoft_scope = @arguments.shift
FoodsoftConfig.select_foodcoop foodsoft_scope if FoodsoftConfig[:multi_coop_install]
```

We'd better `shift` also for singlecoop.

Reply:
yes it only works for multicoop, but not only with my change, was only working with mulricoop since years